pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -117,12 +119,39 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add | framework.Update}},
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
-		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIDriverChange},
```

Comment:
@carlory 

Is there any scenario where CSIDriver creation makes Pod schedulable? We skip a capacity check when it's not found.
https://github.com/kubernetes/kubernetes/blob/8565e375251450a291f0af3c6195c7a5bf890292/pkg/scheduler/framework/plugins/volumebinding/binder.go#L1014-L1018

Reply:
cc @kubernetes/sig-storage-misc 

---

/hold 
QHint implementation depends on whether we really need Creation event or not.

Reply:
It's introduced by https://github.com/kubernetes/kubernetes/pull/100003. 

Let's the primary reviewer of that PR to decide whether to remove the ADD event or not.

/cc @Huang-Wei @ahg-g 

IMO, the ADD event is unnecessary as you said. 

/cc @jsafrane @xing-yang 

Can you also take another look?

Reply:
Good catch, I agree that creation of a CSIDriver cannot make a pending pod schedulable and we don't need Add event here.

This is the only place where CSIDriver instances are checked:
https://github.com/kubernetes/kubernetes/blob/d589a3f866144087e19d092a63b2ef275f2ab912/pkg/scheduler/framework/plugins/volumebinding/binder.go#L1013-L1022
A newly created CSIDriver can make a pod un-schedulable there, but not schedulable.

Reply:
@YamasouA please remove the `ADD` event.

Reply:
@jsafrane @carlory 
Thankyou for your review!
I removed ADD event!!

Reply:
/unhold